### PR TITLE
lib: deprecate `refresh` function

### PR DESCRIPTION
**Pull request by derekstavis**
With great power comes great responsibility. This function is
somewhat low level, and is harmful to shell's user, as it
completely screws up with job control (subprocesses).

This PR officially marks `refresh` as deprecated, moving the
deprecated functionality to omf plugin `compat` quarantine
directory.

### CODE REVIEW

**Comment by bobthecow** [comment]
How does the `</dev/tty` removal affect this?

**Comment by derekstavis** [comment]
In my tests it had no effect, neither on OS X or Linux. I believe `</dev/tty` was added in a try to solve issues with Travis.

File: pkg/omf/functions/compat/refresh.fish [hunk 1]
```diff
@@ -0,0 +1,14 @@
+function refresh -d "(deprecated) Refresh fish session by replacing current process"
+  printf 'warning: function %srefresh%s is deprecated and will be removed soon.\n %s\n' \
+  (set_color -u) (set_color normal) \
+  (status -t)[5] >&2
```
Comment (bobthecow) [inline pkg/omf/functions/compat/refresh.fish]: it's not clear to me what `(status -t)[5]` is intended to return.

Comment (derekstavis) [inline pkg/omf/functions/compat/refresh.fish]: It gets the fifth line of `status -t` :trollface:

Comment (derekstavis) [inline pkg/omf/functions/compat/refresh.fish]: I could split the stack trace and the index expansion to make it clearer what `status -t` output is, or I could use `status --print-stack-trace`.

Comment (bobthecow) [inline pkg/omf/functions/compat/refresh.fish]: Right. But what is it supposed to be?

Comment (derekstavis) [inline pkg/omf/functions/compat/refresh.fish]: It prints where the function was called.

Comment (bobthecow) [inline pkg/omf/functions/compat/refresh.fish]: I can't get it to say anything but standard in.

Comment (derekstavis) [inline pkg/omf/functions/compat/refresh.fish]:
```
function foo
  refresh
end
funcsave foo
foo
```

Comment (derekstavis) [inline pkg/omf/functions/compat/refresh.fish]: This is done so you can discover the offending file

Comment (bobthecow) [inline pkg/omf/functions/compat/refresh.fish]: it always literally just says ` 	called on standard input` for me.

Comment (bobthecow) [inline pkg/omf/functions/compat/refresh.fish]: with some extra spaces at the start.

Comment (derekstavis) [inline pkg/omf/functions/compat/refresh.fish]: for me this is the output of calling function `foo`

```
⋊ > ~/.l/s/omf on deprecate-refresh ● foo
warning: function refresh is deprecated and will be removed soon.
 	called on line 2 of file ~/.config/fish/functions/foo.fish
```

Comment (bobthecow) [inline pkg/omf/functions/compat/refresh.fish]:
```
warning: function refresh is deprecated and will be removed soon.
 	called on standard input
```

Comment (derekstavis) [inline pkg/omf/functions/compat/refresh.fish]: have you saved using `funcsave foo`? :)

Comment (bobthecow) [inline pkg/omf/functions/compat/refresh.fish]: kk. if i call it from a function defined in a file, it works. let's `grep -v` to keep it from talking about standard input and call it a day?

Comment (derekstavis) [inline pkg/omf/functions/compat/refresh.fish]: I don't got your point. You mean hiding the message when it's called in standard input?

Comment (bobthecow) [inline pkg/omf/functions/compat/refresh.fish]: i mean no matter how i call it, it only says "called on standard input" for me. even when that's clearly not the case. i feel like, given this, and given the inability to do anything with that information even if it was true, it makes the most sense to suppress "called on standard input". it's just not useful information :)

File: pkg/omf/functions/compat/refresh.fish [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+  set -q CI
+    and return 0
+
+  type -q omf.core.reload
+    and omf.core.reload
+
```
Comment (oranja) [inline pkg/omf/functions/compat/refresh.fish]: A comment here that `omf.core.reload` calls `exec` and that the lines below are only fallback will be helpful IMO.

Comment (bobthecow) [inline pkg/omf/functions/compat/refresh.fish]: :+1:

**Comment by derekstavis** [comment]
Here we go! Fixed the pointed issues and also rebased to current master! :tada:

**Comment by bobthecow** [comment]
:+1:

**Comment by coderstephen** [comment]
:+1:

**Comment by oranja** [comment]
Looks great to me :+1:

**Comment by bobthecow** [comment]
Maybe add the replacement to the error message?

**Comment by scorphus** [comment]
:+1:

**Pull request merged**